pkg/virtctl/create/instancetype/instancetype.go

```diff
@@ -0,0 +1,286 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2023 Red Hat, Inc.
+ *
+ */
+
+package instancetype
+
+import (
+	"github.com/spf13/cobra"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/rand"
+	v1 "kubevirt.io/api/core/v1"
+	instancetypev1alpha2 "kubevirt.io/api/instancetype/v1alpha2"
+	"sigs.k8s.io/yaml"
+
+	"kubevirt.io/kubevirt/pkg/virtctl/create/params"
+)
+
+const (
+	Instancetype = "instancetype"
+
+	CPUFlag             = "cpu"
+	MemoryFlag          = "memory"
+	GPUFlag             = "gpu"
+	HostDeviceFlag      = "hostdevice"
+	IOThreadsPolicyFlag = "iothreadspolicy"
+	NameFlag            = "name"
+	NamespacedFlag      = "namespaced"
+
+	IOThreadErr   = "IOThread must be of value auto or shared"
+	NameErr       = "name must be specified"
+	DeviceNameErr = "deviceName must be specified"
+)
+
+type createInstancetype struct {
+	name   string
+	cpu    int
+	memory string
+
+	gpus            []string
+	hostDevices     []string
+	ioThreadsPolicy string
+	namespaced      bool
+}
+
+type GPU struct {
+	Name       string `param:"name"`
+	DeviceName string `param:"devicename"`
+}
+
+type HostDevice struct {
+	Name       string `param:"name"`
+	DeviceName string `param:"devicename"`
+}
+
+type optionFn func(*createInstancetype, *instancetypev1alpha2.VirtualMachineInstancetypeSpec) error
+
+var optFns = map[string]optionFn{
+	GPUFlag:             withGPUs,
+	HostDeviceFlag:      withHostDevices,
+	IOThreadsPolicyFlag: withIOThreadsPolicy,
+}
+
+func NewCommand() *cobra.Command {
+	c := createInstancetype{}
+	cmd := &cobra.Command{
+		Use:     Instancetype,
+		Short:   "Create VirtualMachineInstancetype or VirtualMachineClusterInstancetype manifest.",
+		Example: c.usage(),
+		RunE: func(cmd *cobra.Command, _ []string) error {
+			return c.run(cmd)
+		},
+	}
+	cmd.Flags().StringVar(&c.name, NameFlag, c.name, "Specify the name of the Instancetype.")
+	cmd.Flags().IntVar(&c.cpu, CPUFlag, c.cpu, "Specify the count of CPUs of the Instancetype.")
```

Comment:
I suggest making this a `Uint32Var()`, because a negative CPU count does not make sense and you can omit additional input validation by relying on the flag validation.